source/common/network/utility.cc

```diff
@@ -208,7 +218,8 @@ bool Utility::isInternalAddress(const Address::Instance& address) {
   }
 
   if (address.ip()->version() == Address::IpVersion::v4) {
-    // Handle the RFC1918 space for IPV4. Also count loopback as internal.
+    // Handle the RFC1918 space for IPv4, i.e. 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16. Also count
```

Comment:
I thought from previous discussion we need to handle mapped addresses in the IPv6 section below. Won't this create problems for internal services that are trying to infer internal/external from XFF?

Reply:
since we're not pulling existing values from XFF it shouldn't matter. i think i am missing one more check which is to not map internal addresses either, since they could still come from `use_remote_address` (usually at the edge). i will think some more on whether or not hitting that codepath is possible with the current code and configuration permutations. if it is i'll add that logic.

Reply:
We can definitely get an internal address at the edge (e.g., internal load balancer ingress). IMO I think it makes sense to consider a mapped internal address as internal, so I would probably just add the logic in the internal checking code for consistency. But happy to be swayed the other way.

Reply:
after talking with matt offline for a few minutes i'm going to update the logic to check internal and loopback addresses in v4 mapped v6 format

@alyssawilk we also chatted about the possibility of doing mapping at the connection level. let's talk a bit more tomorrow about that.